sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -90,8 +93,18 @@ case class DataSource(
 
   case class SourceInfo(name: String, schema: StructType, partitionColumns: Seq[String])
 
-  lazy val providingClass: Class[_] =
-    DataSource.lookupDataSource(className, sparkSession.sessionState.conf)
+  lazy val providingClass: Class[_] = {
```

Comment:
Since e54caaf188c4575bb6d5e4b4a93d79fce2757547, I tried to avoid changing `DataSource.providingClass` for falling back to V1 when with catalog. 
However, there is a lot test failures including Hive/Streaming. So I tried to pass canonical name of file format to construct the V1 data source, but that will be behavior change.

Another approach is to modify CreateTableCommand, but it is too tricky and it is hard to cover every command.

Hence, in d8f117e26189020a3c7f549f6918c3f9bfd51485 I use `DataSource.providingClass` again, which turn out to be a simple and great solution.